docs/spec/spec.md

```diff
@@ -3274,3 +3345,276 @@ type Product record {
+
+## 11. Experimental Features
+
+This section includes the experimental features in the Ballerina GraphQL package. There _might be_ backward-incompatible changes to these features in future releases. Once a feature is stabilized, it will be graduated as a standard feature.
+
+### 11.1 DataLoader
+
+The Ballerina GraphQL module provides the capability to batch queries to datasources and efficiently cache the retrieved data through the `graphql.dataloader` submodule.
+
+#### 11.1.1 DataLoader API
+
+The `graphql.dataloader` submodule provides the `DataLoader` object, which is used to batch and cache data requests from a data source. The `DataLoader` object type has the following public methods/APIs.
+
+##### 11.1.1.1 The `add` method
+
+This method takes an `anydata` parameter called `key`, which is used to identify the data to be loaded. This method collects and stores the `key` to dispatch a batch operation at a later time. It does not return any values. The following is the method definition of this method.
+
+```ballerina
+public isolated function add(anydata key);
+```
+
+##### 11.1.1.2 The `get` method
+
+This method takes a `key` parameter and retrieves the result for the provided `key`. It performs data binding by examining the type of the assigned variable. In case of failure to retrieve the result or perform data binding, the method returns an error. The following is the method definition of this method.
+
+```ballerina
+public isolated function get(anydata key, typedesc<anydata> 'type = <>) returns 'type|error;
+```
+
+##### 11.1.1.3 The `dispatch` method
+
+This method does not take any parameters and does not return any values. This method is invoked by the GraphQL Engine to dispatch a user-defined batch load function for all the collected keys. For more information about the batch load function, refer to the [Implement the Batch Load Function](#11125-implement-the-batch-load-function) section. The following is the method definition of the `dispatch` method.
+
+```ballerina
+public isolated function dispatch();
+```
+
+##### 11.1.1.4 The `clearAll` method
+
+This method does not take any parameters and does not return any values. The purpose of this method is to clear all the collected keys and cached values from the DataLoader cache. The following is the method definition of the this method.
+
+```ballerina
+public isolated function clearAll();
+```
+
+#### 11.1.2 The DefaultDataLoader
+
+The `DefaultDataLoader` is a built-in implementation of the `DataLoader` object available via the `graphql.dataloader` submodule. Users can use this implementation to batch and cache data loading operations.
+
+#### 11.1.2.1 The `init` Method
+
+The `init` method of the `DefaultDataLoader` object takes a function pointer of type `BatchLoadFunction`. The following is the method definition of the `init` method.
+
+```ballerina
+public isolated function init(BatchLoadFunction loadFunction);
+```
```

Comment:
Let's move this under the DataLoader API section and use an example DataLoader usage here.